### DISCUSSION TITLE
get definition id from http end response

### DISCUSSION THREAD

**Discussion by nurhidayat-agung** [elsa-workflows/elsa-core#5615] [Q&A]
hello elsa
i have worfk flow start with http end point then finish with http response how to get work flow definition id as result http response ?

**Comment by glime-ai** [comment]
To obtain the workflow definition ID as a result in the HTTP response, you can set a variable within your workflow to store the workflow definition ID and then use that variable in the HTTP Response activity to include it in the response body.

Here's a general approach to achieve this:

1. Use the `SetVariable` activity to set a variable with the workflow definition ID. You can use the `WorkflowDefinitionId` property available in the workflow context to get the current workflow definition ID.

2. In the HTTP Response activity, use the variable you set in step 1 to include the workflow definition ID in the response body. You can use an expression to insert the variable into the response content.

Here's an example of how you might configure the activities in your workflow:

```csharp
builder
    .HttpEndpoint("/start")
    // Set a variable with the workflow definition ID
    .SetVariable("WorkflowDefinitionId", "return context.WorkflowDefinitionId;")
    // Other activities...
    .WriteHttpResponse(HttpStatusCode.OK, "return `Workflow Definition ID: ${WorkflowDefinitionId}`;", "text/plain");
```

In this example, the `SetVariable` activity sets a variable named `WorkflowDefinitionId` with the value of the workflow definition ID. The `WriteHttpResponse` activity then uses this variable to include the workflow definition ID in the response content.

Please note that the exact syntax for setting the variable and including it in the response may vary depending on the version of Elsa you are using and the specific details of your workflow. If you need further assistance, please provide more details about your workflow, and I can give you a more precise answer.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
